src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -98,7 +99,13 @@ public void saveMessageInformation(List<Message> messages, PreparedQuery results
 				String text = (String) entity.getProperty("text");
 				long timestamp = (long) entity.getProperty("timestamp");
 				String recipient = (String) entity.getProperty("recipient");
-				Message message = new Message(id, user, text, timestamp, recipient);
+
+				/*Before adding sentiment scores as a feature, there were already messages
+				 without scores. This sets the old sentiment scores to 0 for old messages
+				 */
+				float sentimentScore = entity.getProperty("sentimentScore") == null? (float) 0.0 : ((Double) entity.getProperty("sentimentScore")).floatValue();
```

Comment:
nit: space after null.

Why are you setting one to float and one to Double?

Reply:
They are both float values. The second one is changing the returned value (it returns it as a Double) into a float.